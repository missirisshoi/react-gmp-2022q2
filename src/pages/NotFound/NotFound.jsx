import ErrorBoundary from '../../base/ErrorBoundary';
import Footer from '../../features/Footer';
import styles from './NotFound.module.scss';

const NotFound = () => (
  <ErrorBoundary>
    <div className={styles.not_found}>404. The page is not found.</div>
    <Footer />
  </ErrorBoundary>
);

export default NotFound;
