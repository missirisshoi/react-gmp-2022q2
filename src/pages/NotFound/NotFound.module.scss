.not_found {
  display: flex;
  justify-content: center;
  margin-top: 20vh;
  margin-bottom: 20px;
  min-height: calc(80vh - 90px);
  font-size: 2rem;
  text-align: center;
}
