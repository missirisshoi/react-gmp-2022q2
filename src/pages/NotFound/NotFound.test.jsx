import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NotFound from './NotFound';

describe('404 component', () => {
  test('renders 404', () => {
    render(<NotFound />);
    expect(screen.getByText(/404/)).toBeInTheDocument();
  });

  test('renders logo', () => {
    render(<NotFound />);
    expect(screen.getByText(/netflix/)).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const err = renderer.create(<NotFound />).toJSON();
    expect(err).toMatchSnapshot();
  });
});
