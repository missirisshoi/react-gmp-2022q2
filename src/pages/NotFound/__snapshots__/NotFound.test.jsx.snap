// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 component matches snapshot 1`] = `
Array [
  <div
    className="not_found"
  >
    404. The page is not found.
  </div>,
  <div
    className="footer_wrapper"
  >
    <span
      className="logo_txt"
    >
      <b>
        netflix
      </b>
      roulette
    </span>
  </div>,
]
`;
